splot/_viz_mpl.py

```diff
@@ -0,0 +1,224 @@
+        Moran Scatterplot, LISA cluster map and Choropleth
+    
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> import libpysal.api as lp
+    >>> from libpysal import examples
+    >>> import geopandas as gpd
+    >>> import esda
+    >>> from splot.mpl import plot_local_autocorrelation
+
+    >>> link = examples.get_path('columbus.shp')
+    >>> df = gpd.read_file(link)
+    >>> y = df['HOVAL'].values
+    >>> w = lp.Queen.from_dataframe(df)
+    >>> w.transform = 'r'
+    >>> moran_loc = esda.moran.Moran_Local(y, w)
+    
+    >>> # test with quadrant and mask
+    >>> fig = plot_local_autocorrelation(moran_loc, df, 'HOVAL', p=0.05,
+    ...                                  region_column='POLYID',
+    ...                                  mask=['1', '2', '3'], quadrant=1)
+    >>> plt.show()
+    '''
+    fig, axs = plt.subplots(1, 3, figsize=figsize,
+                            subplot_kw={'aspect':'equal'})
+    #Moran Scatterplot
+    splot.plot.mplot(moran_loc, xlabel='Response', ylabel='Spatial Lag',
+                     title='Moran Scatterplot', p=p, ax=axs[0])
+    axs[0].set_aspect('auto')
+    
+    # Lisa cluster map
+    #TODO: Fix legend_kwds: display boxes instead of points, 
+    lisa_cluster(moran_loc, df, p=p, ax=axs[1], legend=legend,
+                     legend_kwds={'loc': 'upper left', 
+                    'bbox_to_anchor': (0.92, 1.05)})
+    
+    #Choropleth for attribute
+    df.plot(column=attribute, scheme=scheme, cmap=cmap,
+            legend=legend, legend_kwds={'loc': 'upper left',
+                                        'bbox_to_anchor': (0.92, 1.05)},
+            ax=axs[2], alpha=1)
+    axs[2].set_axis_off()
+    
+    #MASKING QUADRANT VALUES
+    if quadrant is not None:
+        #Quadrant masking in Scatterplot
+        mask_angles = {1: 0, 2: 90, 3: 180, 4: 270} #rectangle angles
+        # We don't want to change the axis data limits, so use the current ones
+        xmin, xmax = axs[0].get_xlim()
+        ymin, ymax = axs[0].get_ylim()
+        # We are rotating, so we start from 0 degrees and 
+        # figured out the right dimensions for the rectangles for other angles
+        mask_width = {1: abs(xmax),
+                      2: abs(ymax),
+                      3: abs(xmin),
+                      4: abs(ymin)}
+        mask_height = {1: abs(ymax),
+                      2: abs(xmin),
+                      3: abs(ymin),
+                      4: abs(xmax)}
+        axs[0].add_patch(patches.Rectangle((0,0), width=mask_width[quadrant],
+                                           height=mask_height[quadrant],
+                                           angle=mask_angles[quadrant],
+                                           color='grey', zorder=-1, alpha=0.8))
+        # quadrant selection in maps
+        non_quadrant = ~(moran_loc.q==quadrant)
+        mask_quadrant = df[non_quadrant]
+        df_quadrant = df.iloc[moran_loc.q == quadrant] 
```

Comment:
you can reuse the mask you created above?

Reply:
Yes, thanks that works!